lib/agent0/agent0/hyperdrive/exec/execute_agent_trades.py

```diff
@@ -123,6 +125,20 @@ async def async_execute_single_agent_trade(
             )
         trade_results.append(trade_result)
 
+    # TODO to avoid adding a post action in base policy, we only call post action
+    # if the policy is a hyperdrive policy. Ideally, we'd allow base classes all the
+    # way down
+    if isinstance(agent.policy, HyperdriveBasePolicy):
+        # Calls the agent with the trade results in case the policy needs to do bookkeeping
+        # We copy a subset of fields from the trade results to avoid changing the original
+        # trade result for crash reporting
+        # TODO deepcopy may be inefficient here when copying, e.g., trade_result.agent
+        # If this is the case, we can selectively create a new TradeResult object with a subset
+        # of data
+        trade_result_copy = deepcopy(trade_results)
+        # TODO can't put post_action in agent due to circular import, so we call the policy post_action here
```

Comment:
what is the TODO? I think this looks fine to me, what would you like to change?

Reply:
Ideally, post_action would be an agent function, such that exec doesn't need to access the underlying policy. However, I kept running into circular imports when doing this.